numberformat/diff.emu

```diff
@@ -1018,7 +1018,9 @@
           1. Else,
             1. Let _str_ be ! Number::toString(_x_).
         1. If the grammar cannot interpret _str_ as an expansion of |StringNumericLiteral|, return *NaN*.
-        1. Let _mv_ be the MV, a mathematical value, of ? ToNumber(_str_), as described in <emu-xref href="#sec-runtime-semantics-mv-s"></emu-xref>.
+        1. Let _text_ be ! StringToCodePoints(_str_).
+        1. Let _literal_ be ParseText(_text_, |StringNumericLiteral|).
```

Comment:
Hello @jugglinmike  thanks for your PR, it might be nice to update grammar according to @bakkot [suggestion](https://github.com/tc39/proposal-intl-numberformat-v3/pull/72#issuecomment-965835411 ) and also add this check? 



```suggestion
        1. Let _literal_ be ParseText(_text_, |StringNumericLiteral|).
        1. If _literal_ is a List of errors, let _mv_ be *NaN*.
```

Thanks again 🙏🙏